src/main/java/org/springframework/data/elasticsearch/core/SequenceNumbers.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.core;
+
+/**
+ * @author Roman Puchkovskiy
+ * @since 4.0
+ */
+public class SequenceNumbers {
```

Comment:
I don't think we need this class at all if we change `SeqNoPrimaryTerm`to an immutable data class and put these checks in the factory method.

Reply:
The class has been removed